Allow managementState to be null, default to Removed

jboyd01 opened the pull request:
* bump vendor to pick up latest prereqs which include better support for handling empty managementState
* Add manifest to create cluster ServiceCatalogAPIServer resource with managementStage=Removed
* Add manifest to create ServiceCatalogAPIServer ClusterOperator

Commit:
bump

Commit:
better handle null management state, default to Removed

jmrodri reviewed (approved):
The two manifest files look correct. I see the state is Removed by default which is what you described. The warning in the workload_controller makes sense as well. I did NOT look at the vendor changes in this PR.

jmrodri commented:
/lgtm

bparees commented on manifests/06_service-catalog-apiserver.yaml:
```diff
@@ -0,0 +1,8 @@
+  name: cluster
+spec:
+  logLevel: "Normal"
+  managementState: Removed
```
i'm pretty sure that by doing this, the CVO will start stomping this field back to this value if it's edited (no one will be able to change it to anything else)

if you look at other operators, they create the CR resource programmatically, not via manifests/CVO.

jboyd01 replied:
It has worked on all my tests, but now that you say this, I've always disabled the CVO to get my own operator deployed.  What you are saying makes a ton of sense, thanks Ben, I'll revisit.

bparees commented on pkg/operator/workloadcontroller/workload_controller.go:
```diff
@@ -115,7 +115,9 @@ func (c ServiceCatalogAPIServerOperator) sync() error {
-		c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
+		if len(operatorConfig.Spec.ManagementState) > 0 {
+			c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
+		}
```
why isn't an empty management state also considered to be an unrecognized management state and reported as such?  seems like it's still an error condition in some fashion.

jboyd01 replied:
Perhaps it should be.  This was my understanding in conversations with @derekwaynecarr, but perhaps that was just step along the path to getting Service Catalog into the CVO payload with it disabled.  We are now setting it to an initial state of removed.  I'll revisit this along with the rework above.

Commit:
operator should create cluster ServiceCatalogAPIServer cr (not cvo)

jboyd01 commented:
@bparees thanks for spotting this issue with the creation of of CR in manifests vs having the clusteroperator create it.  I've reworked and addressed the issue with null managementState as well.

bparees commented:
pr description is inaccurate re: null, otherwise lgtm.
